controller/controller.go

```diff
@@ -282,7 +287,23 @@ func (c *Controller) bootWithError(ctx context.Context) error {
 	// users know when to go ahead. Note that mgr.Start below blocks the boot
 	// process until it ends gracefully or fails.
 	{
-		err = ctrl.Watch(&source.Kind{Type: c.newRuntimeObjectFunc()}, &handler.EnqueueRequestForObject{})
+		err = ctrl.Watch(
+			&source.Kind{Type: c.newRuntimeObjectFunc()},
+			&handler.EnqueueRequestForObject{},
+			predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					return matchLabels(c.matchLabels, e.Meta.GetLabels())
+				},
+				DeleteFunc: func(e event.DeleteEvent) bool {
+					return matchLabels(c.matchLabels, e.Meta.GetLabels())
+				},
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					return matchLabels(c.matchLabels, e.MetaNew.GetLabels())
```

Comment:
What is the difference between `Meta` and `MetaNew`? I am wondering why the usage is different for update events.

Reply:
There is `MetaNew` and `MetaOld` in `update` event. if no matching labels in the new schema, we stop send it to a controller.